### PR TITLE
Insecticide reagent, replacement for Pest Spray.

### PR DESCRIPTION
**Pull request by Eneocho**
All other botanical compounds are reagents, change for [consistency]. Jury Rigged to the best of my ability. Tested and mostly works.

Thanks to Kurfursten and B2MTTF for the help.
:cl:
 * rscadd: Added an Insecticide reagent.
 * rscadd: Insecticide reagent poisonous to bugmen and bugmonkeys, to the same degree as plant-b-gone is poisonous to Dionae and dionae nymphs.
 * rscadd: Insecticide bottles on Nutrimax vendor.
 * rscadd: Antitox and Allicin treat Insecticide poisoning.
 * rscadd: Plant-B-Gone and Insecticide on Biogenerator, at 35 Nutriment for 10u, dispensed on inserted beaker.
 * rscadd: Insecticide kills cockroaches, mutroaches, spiderlings and bees.
 * rscdel: Removed Pest Spray from Nutrimax vendor.

### CODE REVIEW

**Comment by drawsstuff** [comment]
I wouldnt remove pest spray from a nutrimax. Vox traders, for example, don't always have a chemistry machine to spare, but are often growing things.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -2472,6 +2472,51 @@
...
+	specheatcap = 4.18
+
+/datum/reagent/toxin/insecticide/reaction_mob(var/mob/living/M, var/method = TOUCH, var/volume)
+
+	if(..())
+		return 1
+	if(iscarbon(M))
+		var/mob/living/carbon/C = M
+		if(!C.wear_mask) //If not wearing a mask
+			C.adjustToxLoss(REM) //4 toxic damage per application, doubled for some reason
```
Comment (boy2mantwicethefam) [inline code/modules/reagents/Chemistry-Reagents.dm:2491]: How'd you get 4 toxin per application? REM is 0.5

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -2472,6 +2472,51 @@
...
+		if(ishuman(M) || ismonkey(M))
```
Comment (boy2mantwicethefam) [inline code/modules/reagents/Chemistry-Reagents.dm]:
```suggestion
		if(ishuman(M))
			var/mob/living/carbon/human/H = M
			if(isinsectoid(H) || istype(M, /mob/living/carbon/monkey/roach)) //Insecticide being poisonous to bugmen, who'd've thunk
				H.adjustToxLoss(10 * REM)
```

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 3]
```diff
@@ -2472,6 +2472,51 @@
...
+			var/mob/living/carbon/human/H = M
+			if(H.dna)
+				if(H.species.flags & isinsectoid(M)) //Insecticide being poisonous to bugmen, who'd've thunk
+					H.adjustToxLoss(10 * REM)
+			if(H.species.flags & isroach(M))
```
Comment (boy2mantwicethefam) [inline code/modules/reagents/Chemistry-Reagents.dm]: isroach doesn't exist and you already gib animals, remove this check

Comment (boy2mantwicethefam) [reply code/modules/reagents/Chemistry-Reagents.dm]: Nevermind, I think you meant to damage the primitive forms of insectoids too
Replace this check with "if(istype(M, /mob/living/carbon/monkey/roach))"

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 4]
```diff
@@ -2472,6 +2472,51 @@
...
+				H.adjustToxLoss(10 * REM)
+					
+/datum/reagent/water/reaction_animal(var/mob/living/simple_animal/M, var/method=TOUCH, var/volume)
+	..()
```
Comment (boy2mantwicethefam) [inline code/modules/reagents/Chemistry-Reagents.dm]: Replace ..() with
```
if(..()) 
  return 1
```

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 1 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm:2491]: I literally copypasted the code from plantbgone. It does the same level of poisoning since it's based on the same recipe (Toxin+Water=PBG, Toxin+Saltwater=Insecticide)

**Comment by Eneocho** [comment]
> 
> 
> I wouldnt remove pest spray from a nutrimax. Vox traders, for example, don't always have a chemistry machine to spare, but are often growing things.

Vox station has 1 chem machine. But I forgot to add the insecticide to the biogenerator. Updating.

**Comment by drawsstuff** [comment]
> > I wouldnt remove pest spray from a nutrimax. Vox traders, for example, don't always have a chemistry machine to spare, but are often growing things.
> 
> Vox station has 1 chem machine. But I forgot to add the insecticide to the biogenerator. Updating.

No, it doesnt. Atleast not box.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 5]
```diff
@@ -2472,6 +2472,54 @@
...
+			C.adjustToxLoss(REM) //4 toxic damage per application, doubled for some reason
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(isinsectoid(H) || istype(M, /mob/living/carbon/monkey/roach)) //Insecticide being poisonous to bugmen, who'd've thunk
+				H.adjustToxLoss(10 * REM)
+			if(H.dna)
+				if(H.species.flags & isinsectoid(M)) //Insecticide being poisonous to bugmen, who'd've thunk
+					H.adjustToxLoss(10 * REM)
+			if(istype(M, /mob/living/carbon/monkey/roach))
+				H.adjustToxLoss(10 * REM)
```
Comment (boy2mantwicethefam) [inline code/modules/reagents/Chemistry-Reagents.dm]: Good Lord I didn't mean it like that

Comment (boy2mantwicethefam) [reply code/modules/reagents/Chemistry-Reagents.dm]:
```suggestion
		if(isinsectoid(C) || istype(C, /mob/living/carbon/monkey/roach)) //Insecticide being poisonous to bugmen, who'd've thunk
			M.adjustToxLoss(10 * REM)
```

**Comment by kilozombie** [comment]
Currently, this only affects two things
It should fulfill the same role as plant-b-gone does for kudzu-- insecticide should demolish bees, roaches, and spiders. Only giant spiders should be immune.
This seems like a very natural and good role and reason for this PR to exist.

**Comment by kilozombie** [comment]
Also, a consistency issue-- the biogen doesn't normally make bottles, it fills the inserted beaker.

**Comment by Eneocho** [comment]
> 
> 
> Also, a consistency issue-- the biogen doesn't normally make bottles, it fills the inserted beaker.

Something something bottles are made of glass, which is made of sand, which can be acquired from grass tiles, that can be made from grass that is grown as a plant. But I get it. Fixed.

**Comment by Eneocho** [comment]
> 
> 
> Currently, this only affects two things
> It should fulfill the same role as plant-b-gone does for kudzu-- insecticide should demolish bees, roaches, and spiders. Only giant spiders should be immune.
> This seems like a very natural and good role and reason for this PR to exist.

I tested the cockroach gib part and it does nothing, apparently. Any idea?

**Comment by kilozombie** [comment]
Would still be nice if it wiped bees off the face of the earth, since they're a real pain to clean up individually whacking them. Same goes with spiders. I noticed that it works on cockroaches though, which is awesome.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 6]
```diff
@@ -2472,6 +2472,56 @@
...
+			C.adjustToxLoss(REM) //4 toxic damage per application, doubled for some reason
+		if(isinsectoid(C) || istype(C, /mob/living/carbon/monkey/roach)) //Insecticide being poisonous to bugmen, who'd've thunk
+			M.adjustToxLoss(10 * REM)
+					
+/datum/reagent/toxin/insecticide/reaction_animal(var/mob/living/simple_animal/M, var/method=TOUCH, var/volume)
+	if(..())
+		return 1
+		
+	if(istype(M,/mob/living/simple_animal/hostile/bigroach))
+		M.gib()
```
Comment (Kurfursten) [inline code/modules/reagents/Chemistry-Reagents.dm]: This has a lot of issues with it.

1. This is really not OOP. This should go inside `reagent_act()` on the associated mob
2. Even if you were going to do it like this, you should've made a list of types and then done `is_type_in_list()` once instead of a series of if statements.
3. This isn't consistent with toxins, which just instantly kills them without gibbing, which makes much more sense for insecticide anyway. This is the only way to kill roaches and such but still leave behind a body, which matters for some applications like grinding roaches for shells.

In short: delete all of this and just add to or replace toxin in their respective `reagent_act()` procs.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 7]
```diff
@@ -2472,6 +2472,56 @@
...
+		
+	if(istype(M,/mob/living/simple_animal/cockroach))
+		M.gib()
+	
+	if(istype(M,/mob/living/simple_animal/bee))
+		M.gib()
+
+	if(istype(M,/mob/living/simple_animal/hostile/giant_spider))
```
Comment (Kurfursten) [inline code/modules/reagents/Chemistry-Reagents.dm]: Instagibbing all spiders, including queens, makes them completely trivial as a threat, not thought through; just get rid of it (you didn't CL this anyway)

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 8]
```diff
@@ -2472,6 +2472,56 @@
...
+		M.gib()
+	
+/datum/reagent/toxin/insecticide/reaction_obj(var/obj/O, var/volume)
+
+	if(..())
+		return 1
+	if(istype(O,/obj/machinery/portable_atmospherics/hydroponics))
```
Comment (Kurfursten) [inline code/modules/reagents/Chemistry-Reagents.dm]: Why even have this if you have `on_plant_life()` below? Confusing why putting pest spray in and out repeatedly with a dropper instead of letting it absorb would allow you to quickly drop the pest level.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 9]
```diff
@@ -2472,6 +2472,56 @@
...
+		var/obj/machinery/portable_atmospherics/hydroponics/tray = O
+		tray.pestlevel -= 2
+		
+/datum/reagent/toxin/insecticide/on_plant_life(obj/machinery/portable_atmospherics/hydroponics/T)
+	..()
```
Comment (Kurfursten) [inline code/modules/reagents/Chemistry-Reagents.dm:2496]: This isn't your fault but I notice the parent proc of this does not work as it should; it handles the sanity but those return 0, meaning you can't add an if statement here to avoid runtimes if it has no holder somehow.

You don't need to change this though, since it seems very unlikely to happen. Just pointing it out for general notice.

**Review by Kurfursten — CHANGES_REQUESTED** [review]

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 6 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm]: I am a codelet and literally jury rigged this to the best of my understanding, please give me steps to do or the specific changes.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 7 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm]: I didn't? Fuck me, I didn't notice, I must've blanked while I was updating the CL. I was trying for it to instagib the small spiders. Could you tell me how to swap it to instagib the small baby spiders but not the normal or queen spiders?

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 8 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm]: Jury rigged. Should I delete it?

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 9 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm:2496]: GOD I wish I knew what that meant codewise.

**Comment by Kurfursten** [comment]
Remind me tomorrow, just closed on a house

**Comment by Eneocho** [comment]
> 
> 
> Remind me tomorrow, just closed on a house

Aight. Good luck.

**Comment by Eneocho** [comment]
> 
> 
> Remind me tomorrow, just closed on a house

Can you help me now or are you still busy?

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 6 above)
Comment (Kurfursten) [reply code/modules/reagents/Chemistry-Reagents.dm]: go to each mob's file (e.g. `simple_animal/roach.dm`) and look for a proc called `reagent_act()` (L292 in the case of roach.dm) (ctrl L to toggle line numbers, ctrl G to go to line).  In the case of roaches, your job is 98% already done, just add `, INSECTICIDE` after `TOXIN` at line 299.

Repeat this inside the reagent_act() proc for bees and spiderlings, then delete this whole series of if statements.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 7 above)
Comment (Kurfursten) [reply code/modules/reagents/Chemistry-Reagents.dm]: Do what I said above regarding reagent_act() inside `simple_animal/hostile/giant_spider/spiderling.dm`; unlike the other 2, it doesn't have an existing proc, so you need to make one -- make sure when you path it that it includes the full path (e.g.: `mob/living/simple_animal/hostile/giant_spider/spiderling/reagent_act(id, method, volume)` )

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 8 above)
Comment (Kurfursten) [reply code/modules/reagents/Chemistry-Reagents.dm]: Yes

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 9 above)
Comment (Kurfursten) [reply code/modules/reagents/Chemistry-Reagents.dm:2496]: Don't worry about this since it's above your level

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 6 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm]: Done.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 7 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm]: Fixed, I think.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 8 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm]: Deleted.

**Comment by Eneocho** [comment]
Should spiders and spiderqueens be affected by insecticide? Poisoned or something. Or should they have immunity?

An idea would be:
Spiderlings die (coded).
Spiders get heavily poisoned.
Queens are largely immune (1/20th the damage a normal spider would get?)

**Comment by Eneocho** [comment]
No idea how to fix the gore part.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 9 above)
Comment (Eneocho) [reply code/modules/reagents/Chemistry-Reagents.dm:2496]: welp

File: code/modules/mob/living/simple_animal/hostile/giant_spider/spiderling.dm [hunk 10]
```diff
@@ -55,6 +55,17 @@
...
+/mob/living/simple_animal/giant_spider/spiderling/reagent_act(id, method, volume)
+	if(isDead())
+		return
+
+	.=..()
+
+	switch(id)
+		if(INSECTICIDE)
+			if(method != INGEST)
+				death(gore = 0)
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/hostile/giant_spider/spiderling.dm]: `death(FALSE)`

File: code/modules/mob/living/simple_animal/hostile/bigroach.dm [hunk 11]
```diff
@@ -124,3 +124,14 @@
...
+/mob/living/simple_animal/hostile/bigroach/reagent_act(id, method, volume)
+	if(isDead())
+		return
+
+	.=..()
+
+	switch(id)
+		if(INSECTICIDE)
+			if(method != INGEST)
+				death(gore = 0)
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/hostile/bigroach.dm]: just `death(FALSE)` here is fine

**Review by Kurfursten — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/bigroach.dm (hunk 11 above)
Comment (Eneocho) [reply code/modules/mob/living/simple_animal/hostile/bigroach.dm]: Done.

File: code/modules/mob/living/simple_animal/hostile/giant_spider/spiderling.dm (hunk 10 above)
Comment (Eneocho) [reply code/modules/mob/living/simple_animal/hostile/giant_spider/spiderling.dm]: Fixed.

**Comment by DeityLink** [comment]
> rscadd: Plant-B-Gone and Insecticide on Biogenerator, at 35 Nutriment for 10u, dispensed on inserted beaker.

Wait does that mean we finally have a way to produce more chem sprayers?

**Comment by Kurfursten** [comment]
Chem sprayers require illegal tech and are printed at the protolathe
What you may be thinking of is spray bottles, and no, this dispenses in the inserted beaker

**Comment by DeityLink** [comment]
Yeah I meant sprays, and oh I see.
